sdk/include/opentelemetry/sdk/trace/batch_span_processor.h

```diff
@@ -141,22 +141,16 @@ class BatchSpanProcessor : public SpanProcessor
   void DrainQueue();
 
 #ifdef ENABLE_ASYNC_EXPORT
-  struct AsyncExportData
-  {
-    nostd::span<std::unique_ptr<Recordable>> recordables;
-  };
-
   struct ExportDataStorage
   {
-    std::unordered_map<AsyncExportData *, std::unique_ptr<AsyncExportData>> running_async_exports;
-    std::list<std::unique_ptr<AsyncExportData>> garbage_async_exports;
+    std::queue<size_t> export_ids;
+    std::vector<bool> export_ids_flag;
   };
   std::shared_ptr<ExportDataStorage> export_data_storage_;
 
-  bool CleanUpGarbageAsyncData();
-
   const bool is_export_async_;
   const size_t max_export_async_;
+  static constexpr int kInvalidExportId = 0;
```

Comment:
Maybe we can use `constexpr size_t kInvalidExportId = static_cast<size_t>(-1);` here? Which is just like `std::string::npos`. And some compiler will warn comparison between sign and unsigned interger, so we should use size_t for better.

Reply:
Agreed.